vsintegration/src/vs/FsPkgs/FSharp.Project/Common.Source.CSharp/Project/TypeProviderContainingReferenceNode.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Open Technologies, Inc.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.VisualStudio.FSharp.ProjectSystem
+{
+    [CLSCompliant(false), ComVisible(true)]
+    public abstract class TypeProviderContainingReferenceNode : ReferenceNode
```

Comment:
convert class `TypeProviderContainingReferenceNode` to interface + a static function for imageindex?
there are already lot of classes, i understand like this the behavior is shared but not worth a class